pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/HadoopPinotFS.java

```diff
@@ -139,6 +205,20 @@ public long length(URI fileUri) throws IOException {
     return retArray;
   }
 
+  private FileStatus[] listStatus(Path path, boolean recursive) throws IOException {
+    List<FileStatus> fileStatuses = new ArrayList<>();
+    FileStatus[] files = _hadoopFS.listStatus(path);
+    for (FileStatus file : files) {
+      fileStatuses.add(file);
+      if (file.isDirectory() && recursive) {
+        List<FileStatus> subFiles = Arrays.asList(listStatus(file.getPath(), true));
+        fileStatuses.addAll(subFiles);
+      }
+    }
+    FileStatus[] fileStatusesArr = new FileStatus[fileStatuses.size()];
```

Comment:
Why can't we return the array-list here?

Reply:
The reason is that I'm not sure whether this API will be exposed in the future. Right now all the `listStatus` from `Hadoop FileSystem` return an FileStatus array. So I think it's best to follow the convention.